test/common/upstream/health_check_fuzz.cc

```diff
@@ -0,0 +1,92 @@
+#include "test/common/upstream/health_check_fuzz.h"
+
+#include <chrono>
+
+#include "test/common/upstream/utility.h"
+#include "test/fuzz/utility.h"
+
+namespace Envoy {
+namespace Upstream {
+
+void HealthCheckFuzz::allocHealthCheckerFromProto(
+    const envoy::config::core::v3::HealthCheck& config) {
+  health_checker_ = std::make_shared<TestHttpHealthCheckerImpl>(
+      *cluster_, config, dispatcher_, runtime_, random_,
+      HealthCheckEventLoggerPtr(event_logger_storage_.release()));
+  ENVOY_LOG_MISC(trace, "Created Test Health Checker");
+}
+
+void HealthCheckFuzz::initialize(test::common::upstream::HealthCheckTestCase input) {
+  allocHealthCheckerFromProto(input.health_check_config());
+  addCompletionCallback();
+  if (input.second_host()) {
+    cluster_->prioritySet().getMockHostSet(0)->hosts_ = {
+        Upstream::makeTestHost(cluster_->info_, "tcp://127.0.0.1:80"),
+        Upstream::makeTestHost(cluster_->info_, "tcp://127.0.0.1:81")};
+  } else {
+    cluster_->prioritySet().getMockHostSet(0)->hosts_ = {
+        Upstream::makeTestHost(cluster_->info_, "tcp://127.0.0.1:80")};
+  }
+  expectSessionCreate();
+  expectStreamCreate(0);
+  if (input.second_host()) {
+    ENVOY_LOG_MISC(trace, "Created second host.");
+    second_host_ = true;
+    expectSessionCreate();
+    expectStreamCreate(1);
+  }
+  health_checker_->start();
+  replay(input);
+}
+
+void HealthCheckFuzz::respondHeaders(test::fuzz::Headers headers, absl::string_view status,
+                                     bool respond_on_second_host) {
+  std::unique_ptr<Http::TestResponseHeaderMapImpl> response_headers =
+      std::make_unique<Http::TestResponseHeaderMapImpl>(
+          Fuzz::fromHeaders<Http::TestResponseHeaderMapImpl>(headers, {}, {}));
+
+  // If Fuzzer created a status header - replace it with validated status
```

Comment:
is the statement "if fuzzer created a status header" happening?

Reply:
Yes, but rarely. Removed comment